### ISSUE TITLE
Image flash when switching tabs

### ISSUE THREAD

**Issue by brzzdev** [kean/Nuke#709]
Hi,

I've been using Nuke for a while and I love it, but I noticed that there's a noticible flash when switching between tabs

https://github.com/kean/Nuke/assets/15687450/3a3afeb2-01fc-49b5-b69d-4ea53113d6ec

as compared to say Kingfisher

https://github.com/kean/Nuke/assets/15687450/e2bbb330-dd7f-48c0-b318-b419b819a848

Here's the code for both (url: URL, color: Color):

Nuke:

```
@MainActor @ViewBuilder
private func NukeImageView() -> some View {
    LazyImage(url: url) { state in
        if let image = state.image {
            image
                .resizable()
                .aspectRatio(contentMode: .fill)
        } else {
            color
        }
    }
}
```

Kingfisher:

```
@ViewBuilder
private func KFImageView() -> some View {
    KFImage(url)
        .contentConfigure { image in
            image
                .resizable()
                .aspectRatio(contentMode: .fill)
        }
        .placeholder {
            color
        }
}
```

I see no flash with the built-in AsyncImage either:

```
@ViewBuilder
private func ImageView() -> some View {
    AsyncImage(url: url) { image in
        image
            .resizable()
            .aspectRatio(contentMode: .fill)
    } placeholder: {
        color
    }
}
```

**Comment by kean** [comment]
Hi, which version of Nuke are you using?

**Comment by PaulDoesDev** [comment]
The latest, 12.1.1, but definitely saw it earlier.

Happy to go back and see if it's all of 12, if that's useful. Don't think that code would work pre-12 though.

**Comment by AgapovOne** [comment]
I think we have the same problem.

@PaulDoesDev do you have a demo project to share?

**Comment by swasta** [comment]
We experience the same behavior with the 12.1.2 Nuke. onAppear() gets triggered inside LazyImage and the check `guard viewModel.cachedResponse == nil` is true for some reason, although I would expect cachedResponse not be nil. Besides, if I call `po pipeline.cache[context!.request]` on the same line, some ImageContainer gets printed, which looks suspicious to me because I read it as "There's an image cached in-memory". Besides, if I scroll the grid with images a bit, trigger cachedResponse assignment, and only then switch the tabs — there's no flash

**Comment by PaulDoesDev** [comment]
Quick and dirty [demo app](https://github.com/PaulDoesDev/NukeFlash)

**Comment by kean** [comment]
Thanks, @PaulDoesDev. Are there any additional steps needed to repro it? When I switch tabs, it doesn't blink.

I tested on iOS 16.4 and iOS 17.0.

https://github.com/kean/Nuke/assets/1567433/c3e50814-ea64-40ae-a565-21e4a9d9514e

**Comment by PaulDoesDev** [comment]
In the Simulator, I can get it to happen if I keep switching back and forth between tabs repeatedly. On device though, it happens on every tab switch.

iPhone 14 Pro - 17.0 beta2
iPhone 12 - 16.5.1

**Comment by kean** [comment]
Yes, I can see it on the device. It seems like a dropped frame, which is weird because `LazyImage` has `performCacheLookupIfNeeded` that was specifically added to make sure cached images are available on the first draw. I haven't looked further into it, but `performCacheLookupIfNeeded` and `isCacheLookupNeeded` are the main suspect. I thin it could be simplified, and `LazyImage` could perform cache lookups on every redraw.

**Comment by AgapovOne** [comment]
What bothers us in our project is the animation. When I even assign a default animation to lazy image in demo project

`LazyImage(url: URL(string: image), transaction: .init(animation: .default))`

I get reassigning through red placeholder on every tab change.

I would expect transaction to happen only after loading step, not initial assignment.

**Comment by kean** [comment]
Fixed in [12.1.3](https://github.com/kean/Nuke/releases/tag/12.1.3).
